test/extended/networking/egressip.go

```diff
@@ -175,6 +176,54 @@ var _ = g.Describe("[sig-network][Feature:EgressIP][apigroup:operator.openshift.
 				o.Expect(err).NotTo(o.HaveOccurred())
 			}
 		}
+		g.By(fmt.Sprintf("The test pods namespace %s will be connected to network named %q with network topology %q", egressIPNamespace, netConfig.name, netConfig.topology))
+		// Configure the user defined network for the test namespace if required
+		if isClusterDefaultNetwork(netConfig.networkName) {
+			return
+		}
+		// Skip If CNI is OpenShift SDN and "primary network" is a secondary network
+		if networkPluginName() == OpenshiftSDNPluginName && !isClusterDefaultNetwork(netConfig.networkName) {
+			skipper.Skipf("OpenShift SDN does not support EgressIP on secondary networks")
+		}
+		// Skip if cluster doesn't contain the tech preview feature gate
+		if !exutil.IsTechPreviewNoUpgrade(context.Background(), oc.AdminConfigClient()) {
+			skipper.Skipf("User defined networks require feature gate tech preview")
+		}
+		// configure UDN in the egressIPNamespace namespace
+		nadClient, err := nadclient.NewForConfig(f.ClientConfig())
+		o.Expect(err).NotTo(o.HaveOccurred(), "network attachment client must be available")
+		netConfig.cidr = correctCIDRFamily(oc, "30.10.0.0/16", "2014:100:200::0/60")
+		g.By(fmt.Sprintf("User Defined Network CIDR is %q", netConfig.cidr))
+		// egressIPNamespace & externalNamespace will be connected to the same network
+		netConfig := newNetworkAttachmentConfig(netConfig)
+		g.By(fmt.Sprintf("Creating the network %s with network topology %s and connecting the network to Namespace %s",
+			netConfig.name, netConfig.topology, externalNamespace))
+		netConfig.namespace = externalNamespace
+		_, err = nadClient.NetworkAttachmentDefinitions(externalNamespace).Create(
+			context.Background(),
+			generateNAD(netConfig),
+			metav1.CreateOptions{},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred(), "creation of network must succeed")
+		netConfig.namespace = egressIPNamespace
+		g.By(fmt.Sprintf("Creating the network %s with network topology %s and connecting the network to Namespace %s",
+			netConfig.name, netConfig.topology, egressIPNamespace))
+		_, err = nadClient.NetworkAttachmentDefinitions(egressIPNamespace).Create(
+			context.Background(),
+			generateNAD(netConfig),
+			metav1.CreateOptions{},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred(), "creation of network must succeed")
+		g.By("Ensure the network is configured on a set of worker Nodes")
+		for _, workerNodeName := range workerNodesOrderedNames {
+			var isNetProvisioned bool
+			o.Eventually(func() bool {
+				isNetProvisioned, err = isNetworkProvisioned(oc, workerNodeName, netConfig.networkName, netConfig.topology)
+				return err == nil && isNetProvisioned
+			}).WithPolling(time.Second).WithTimeout(50 * time.Second).Should(
+				o.BeTrueBecause(fmt.Sprintf("the network must be ready before creating workloads for node %s, last err: %v", workerNodeName, err)),
```

Comment:
Nit: you may skip explicit `fmt.Sprintf `because `o.BeTrueBecause` is also doing the same.